StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1202ElementsMustBeOrderedByAccess.cs

```diff
@@ -126,79 +126,60 @@ private static void HandleCompilationUnit(SyntaxNodeAnalysisContext context)
+            HandleMemberList(context, typeDeclaration.Members);
         }
 
-        private static void HandleMemberList(SyntaxNodeAnalysisContext context, IList<MemberDeclarationSyntax> members, string memberType)
+        private static void HandleMemberList(SyntaxNodeAnalysisContext context, SyntaxList<MemberDeclarationSyntax> members)
         {
-            for (int i = 0; i < members.Count - 1; ++i)
-            {
-                var currentAccessLevel = GetAccessLevel(members[i]);
-                if (currentAccessLevel == AccessLevel.NotSpecified)
-                {
-                    continue;
-                }
-
-                var nextAccessLevel = GetAccessLevel(members[i + 1]);
-                if (nextAccessLevel == AccessLevel.NotSpecified)
-                {
-                    continue;
-                }
+            var previousMemberStatic = true;
+            var previousMemberReadonly = true;
+            var previousSyntaxKind = SyntaxKind.None;
+            var previousAccessLevel = AccessLevel.NotSpecified;
 
-                if (currentAccessLevel > nextAccessLevel)
+            foreach (var member in members)
+            {
+                var currentModifiers = member.GetModifiers();
+                var currentMemberStatic = currentModifiers.Any(SyntaxKind.StaticKeyword);
+                var currentMemberReadonly = currentModifiers.Any(SyntaxKind.ReadOnlyKeyword);
+                var currentSyntaxKind = member.Kind();
+                currentSyntaxKind = currentSyntaxKind == SyntaxKind.EventFieldDeclaration ? SyntaxKind.EventDeclaration : currentSyntaxKind;
+                var currentAccessLevel = GetAccessLevel(currentModifiers);
+
+                if (previousAccessLevel != AccessLevel.NotSpecified
+                    && currentAccessLevel != AccessLevel.NotSpecified
+                    && currentSyntaxKind == previousSyntaxKind
+                    && currentMemberStatic == previousMemberStatic
+                    && currentMemberReadonly == previousMemberReadonly
+                    && currentAccessLevel < previousAccessLevel)
                 {
                     context.ReportDiagnostic(
-                        Diagnostic.Create(Descriptor,
-                        NamedTypeHelpers.GetNameOrIdentifierLocation(members[i + 1]),
-                        UpperAccessLevelNames[nextAccessLevel],
-                        memberType,
-                        LowerAccessLevelNames[currentAccessLevel]));
+                    Diagnostic.Create(Descriptor,
+                    NamedTypeHelpers.GetNameOrIdentifierLocation(member),
+                    UpperAccessLevelNames[currentAccessLevel],
+                    currentMemberStatic ? " static" : string.Empty,
+                    MemberNames[currentSyntaxKind],
+                    LowerAccessLevelNames[previousAccessLevel]));
```

Comment:
each level should be indented

    context.ReportDiagnostic(
        Diagnostic.Create(
            Descriptor,
            NamedTypeHelpers.GetNameOrIdentifierLocation(member),
    ...